runtime/oti/jilconsts_nasm.inc

```diff
@@ -0,0 +1,194 @@
+
+; temporary workaround!!!
+
+; this is only temporarily placed here to have the nasm files being able to include a replacement for "jilconsts.inc"
+; Currently, the jilconsts.inc is in masm format, so until that is addressed, this is a temporary placeholder
+
+
+%assign ASM_J9VM_JIT_NEW_DUAL_HELPERS 1
+%assign ASM_J9VM_ENV_LITTLE_ENDIAN 1
+%assign ASM_J9VM_ENV_DATA64 1
+%assign ASM_J9VM_INTERP_COMPRESSED_OBJECT_HEADER 1
+%assign ASM_J9VM_THR_LOCK_NURSERY 1
+%assign ASM_J9VM_INTERP_SMALL_MONITOR_SLOT 1
+%assign ASM_J9VM_GC_TLH_PREFETCH_FTA 1
+%assign ASM_J9VM_GC_DYNAMIC_CLASS_UNLOADING 1
+%assign J9TR_cframe_sizeof 280
+%assign J9TR_cframe_vmStruct 0
+%assign J9TR_cframe_machineBP 8
+%assign J9TR_cframe_jitGPRs 16
+%assign J9TR_cframe_jitFPRs 144
+%assign J9TR_cframe_rax 16
+%assign J9TR_cframe_rbx 24
+%assign J9TR_cframe_rcx 32
+%assign J9TR_cframe_rdx 40
+%assign J9TR_cframe_rdi 48
+%assign J9TR_cframe_rsi 56
+%assign J9TR_cframe_rbp 64
+%assign J9TR_cframe_rsp 72
+%assign J9TR_cframe_r8 80
+%assign J9TR_cframe_r9 88
+%assign J9TR_cframe_r10 96
+%assign J9TR_cframe_r11 104
+%assign J9TR_cframe_r12 112
+%assign J9TR_cframe_r13 120
+%assign J9TR_cframe_r14 128
+%assign J9TR_cframe_r15 136
+%assign J9TR_runtimeFlags_PatchingFenceRequired 67108864
+%assign J9TR_runtimeFlags_PatchingFenceType 134217728
+%assign J9TR_VMThread_machineSP 2392
+%assign J9TR_machineSP_machineBP 8
+%assign J9TR_machineSP_vmStruct 0
+%assign J9TR_VMThreadCurrentException 72
+%assign J9TR_VMThread_floatTemp1 264
+%assign J9TR_VMThread_floatTemp2 272
+%assign J9TR_VMThread_floatTemp3 280
+%assign J9TR_VMThread_floatTemp4 288
+%assign J9TR_VMThread_stackOverflowMark 80
+%assign J9TR_VMThread_heapAlloc 96
+%assign J9TR_VMThread_heapTop 104
+%assign J9TR_VMThread_tlhPrefetchFTA 112
+%assign J9TR_VMThread_publicFlags 152
+%assign J9TR_VMThread_privateFlags 424
+%assign J9TR_VMThread_privateFlags2 2336
+%assign J9TR_VMThreadJavaVM 8
+%assign J9TR_VMThread_javaVM 8
+%assign J9TR_VMThread_sp 32
+%assign J9TR_VMThread_pc 40
+%assign J9TR_VMThread_literals 48
+%assign J9TR_VMThread_arg0EA 16
+%assign J9TR_VMThread_codertTOC 344
+%assign J9TR_VMThread_debugEventData1 496
+%assign J9TR_VMThread_debugEventData2 504
+%assign J9TR_VMThread_debugEventData3 512
+%assign J9TR_VMThread_debugEventData4 520
+%assign J9TR_VMThread_debugEventData5 528
+%assign J9TR_VMThread_debugEventData6 536
+%assign J9TR_VMThread_debugEventData7 544
+%assign J9TR_VMThread_debugEventData8 552
+%assign J9TR_VMThread_tempSlot 248
+%assign J9TR_VMThread_jitReturnAddress 256
+%assign J9TR_VMThread_returnValue 296
+%assign J9TR_VMThread_returnValue2 304
+%assign J9TR_VMThread_entryLocalStorage 616
+%assign J9TR_VMThread_stackWalkState 608
+%assign J9TR_J9StackWalkState_restartPoint 224
+%assign J9TR_JavaVMJitConfig 6384
+%assign J9TR_JavaVM_runtimeFlags 224
+%assign J9TR_JavaVM_cInterpreter 184
+%assign J9TR_JavaVM_bytecodeLoop 192
```

Comment:
This is very fragile and will break the first time someone modifies one of these structures.  We need to have a story that generates the nasm form of this file in addition to the masm form.

Reply:
Yup, this is only temporary to get the builds to pass. I am working on auto-generating the NASM variant, using values from the generated jilconsts.inc file.